src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs

```diff
@@ -135,7 +136,12 @@ private static PortableExecutableReference MakeAssemblyMetadata(ModuleMetadata m
                         // or if the module has not been loaded yet. The value will be null
                         // if the name was ambiguous.
                         ModuleMetadata module;
-                        if (modulesByName.TryGetValue(name, out module) && (module != null))
+                        if (!modulesByName.TryGetValue(name, out module))
+                        {
+                            // AssemblyFile names may contain file information (".dll", etc).
+                            modulesByName.TryGetValue(Path.GetFileNameWithoutExtension(name), out module);
```

Comment:
Do we have to worry about false positives?  What if we can't find "A.B.C", but "A.B" is loaded?  Do we have some reason to believe that "A.B.C" will only be looked up if it's available?